### PR TITLE
CI for performance tests

### PR DESCRIPTION
**Pull request by lostmsu**
### What does this implement/fix? Explain your changes.

Runs performance tests in CI, comparing build with new changes against the latest (2.3 as of now) NuGet release.

### Does this close any currently open issues?

#886

### Checklist

Nothing applicable.

### CODE REVIEW

**Comment by lostmsu** [comment]
@filmor  if you are to merge this, I recommend squashing the commit, as there are many intermediate changes, which are insignificant on their own.

File: src/perf_tests/Python.PerformanceTests.csproj [hunk 1]
```diff
@@ -3,12 +3,19 @@
...
+    <OutputPath>bin\</OutputPath>
 
     <IsPackable>false</IsPackable>
+
+    <Platforms>x64;x86</Platforms>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="BenchmarkDotNet" Version="0.11.5" />
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies.net461" Version="1.0.0-preview.2">
```
Comment (filmor) [inline src/perf_tests/Python.PerformanceTests.csproj]: Any particular reason for referencing net461 explicitly? I was under the impression that the package selects the correct one without a version.

Comment (lostmsu) [reply src/perf_tests/Python.PerformanceTests.csproj]: I am unsure if selection happens before downloading or after. If it has to download all packages first, then it should shorten build time. I can look it up, but I hope we will get rid of this soon, as it is only needed for pythonnet-2.3 packages, that don't support .NET Standard.

**Comment by lostmsu** [comment]
OK, test is passing, and I can see in [AppVeyor logs for 3.5 x64 xplat build](https://ci.appveyor.com/project/pythonnet/pythonnet/builds/29033240/job/535gmdinq8hnvbul#L2050), that performance is checked.

BTW, 3.5 x64 xplat is the only configuration where perf test runs, as it is the only configuration supported by the published NuGet package.

**Comment by lostmsu** [comment]
I guess this boils down to how much flakiness we can tolerate in perf tests in CI. Letting a bit more wiggle room in the performance goal should make it very unlikely to fail randomly.

File: src/perf_tests/Python.PerformanceTests.csproj (hunk 1 above)
Comment (lostmsu) [reply src/perf_tests/Python.PerformanceTests.csproj]: Fixed

**Pull request merged**